app/models/order.rb

```diff
@@ -18,4 +18,27 @@ class Order < ApplicationRecord
 
   scope :with_status, ->(status){where(status:) if status.present?}
   scope :for_user, ->(user_id){where(user_id:) if user_id.present?}
+  scope :sort_by_payment_method, lambda {|direction = :asc|
+    joins(:payment_method)
+      .order(Arel.sql("FIELD(payment_methods.payment_method,
+      'credit_card', 'paypal', 'bank_transfer') #{direction.to_s.upcase}"))
+  }
+
+  ransack_alias :search,
+                :status_or_reason_or_payment_method_id_or_user_id_cont
```

Comment:
làm như này để làm gì em. Sao mà thích thập cẩm thế

Reply:
em sort theo enum, em ép kiểu và string và join vì payment_method ở bảng khác

Reply:
Search theo đk and ấy đừng cứ mãi search theo đk or

Reply:
@anlvq-1058 anh noi cho rank_alisas á anh ; em làm theo yêu cầu số 7 mà 
7	Có 1 query group sử dụng OR thay vì mặc định AND